test/app-luatest/gh_4317_console_backslash_test.lua

```diff
@@ -0,0 +1,57 @@
+local t = require('luatest')
+local g = t.group()
+
+local result_str =  [[tarantool> \set c \
+---
+- true
+...
+
+tarantool> local a = 0 \
+         > for i = 1, 10 do
+         > a = a + i
+         > end \
+         > print(a)
+55
+---
+...
+
+tarantool> ]]
+
+local TARANTOOL_PATH = arg[-1]
+
+g.test_using_backslash_on_local_console = function()
+    local tarantool_command = [[\\set c \\\nlocal a = 0 \\\nfor i = 1, 10 do\na = a + i\nend \\\nprint(a)\n]]
+
+    local cmd = ("printf '%s' | %s -i 2>/dev/null"):format(tarantool_command, TARANTOOL_PATH)
+    local fh = io.popen(cmd, 'r')
+
+    -- Readline on CentOS 7 produces \e[?1034h escape sequence before tarantool> prompt, remove it.
```

Comment:
Please add `XXX:` prefix to the comment, since this is a specific spot fixed with a hack.

Reply:
Done.